segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java

```diff
@@ -989,6 +989,18 @@ private void shutdownWhenStopped(Service component, String componentName) {
         return this.durableLog.add(op, priority, timeout).thenApply(ignored -> op.getStreamSegmentId());
     }
 
+    private CompletableFuture<Boolean> pinSegment(long segmentId, String segmentName, boolean pin, Duration timeout) {
+        Preconditions.checkNotNull(segmentName, "SegmentName cannot be null.");
+        Preconditions.checkArgument(segmentId != ContainerMetadata.NO_STREAM_SEGMENT_ID, "Segment must have a valid id.");
+
+        UpdateableSegmentMetadata segmentMetadata = this.metadata.getStreamSegmentMetadata(segmentId);
+        if (segmentMetadata != null && pin) {
+            segmentMetadata.markPinned();
+            return CompletableFuture.completedFuture(true);
+        }
+        return CompletableFuture.completedFuture(false);
```

Comment:
looks like we can get rid of parameter `boolean pin`.
It's never called with false value.

Reply:
Could be, but this is perhaps more explicit (just kept the structure used for this flag in other places).